tests/Unit/API/Google/AdsCampaignTest.php

```diff
@@ -70,6 +70,11 @@ public function setUp() {
 
 	public function test_get_campaigns_empty_list() {
 		$this->generate_ads_campaign_query_mock( [], [] );
+
+		// Method search() only being called by AdsCampaignQuery
+		$this->service_client->expects( $this->once() )
+			->method( 'search' );
```

Comment:
Since `service_client` is part of HelperTrait I think it would be cleaner to create a separate helper function `generate_ads_campaign_query_mock_with_no_campaigns`. A comment can be added there that it only expects one of the queries to run.

Reply:
Thanks, also updated in 40fafee.